packages/frontend/src/screens/app/screens/tournaments/screens/tournament/screens/match/match-view.tsx

```diff
@@ -14,24 +16,54 @@ export interface MatchViewMatch {
 	secondTeam: Team;
 	winner: Team | null;
 	contested: boolean;
+	tournament: { id: string; owner: { id: string } };
+	finalized: boolean;
 }
 
 interface Props {
 	reportWin: (teamId: string) => void;
 	reportContested: (contested: boolean) => void;
+	reportFinalized: () => void;
 	match: MatchViewMatch;
 	loading: boolean;
+	tournament: {
+		id: string;
+		owner: { id: string };
+	};
 }
 
-const MatchView: React.FC<Props> = ({ reportWin, reportContested, match, loading }) => {
+const MatchView: React.FC<Props> = ({
+	reportWin,
+	reportContested,
+	reportFinalized,
+	match,
+	loading,
+	tournament,
+}) => {
+	function firstTeamRetcon() {
```

Comment:
What is Retcon?

Reply:
https://www.urbandictionary.com/define.php?term=retcon
Retcon is short for Retroactive Continuity. Being a popular term in DnD and boardgames. Mean that we change the continuity to something else. Here I have used in the sense that we change what team has won. We can change the function name if so desired.